impl/src/main/java/org/jboss/weld/AbstractCDI.java

```diff
@@ -128,4 +136,11 @@ protected String getCallingClassName() {
         return instanceCache.getValue(BeanManagerProxy.unwrap(getBeanManager()));
     }
 
+    /**
+     * Check whether container is running
+     */
+    protected void checkState(){
```

Comment:
Why not just make this `abstract`? So that subclasses (WeldContainer) have to implement this.
`protected abstract void checkState();`